pkg/controller/clone-controller.go

```diff
@@ -46,6 +46,8 @@ const (
 	AnnCloneOf = "k8s.io/CloneOf"
 	// AnnCloneToken is the annotation containing the clone token
 	AnnCloneToken = "cdi.kubevirt.io/storage.clone.token"
+	// AnnCloneSourcePodFailed is an annotation used to specify that the source clone pod failed
+	AnnCloneSourcePodFailed = "cdi.kubevirt.io/storage.source.clone.pod.failed"
```

Comment:
I would not call it failed, its just not running, so I would use several of the same annotations associated with the target, running/reason/message but just source versions. That way we can take source values and the target values and in the DV controller build a coherent message for the user.

Reply:
hmmm after thinking about it, I don't know if this option is better than using the current annotation: The pod actually fails, in this case, for quota limitations (this annotation is added in the `handleFailedPod` function).

The annotation is used to let the upload controller know that the source-clone pod failed (it did), but it has no direct relation with the source PVC. And, since the source PVC is still running (the condition is changed in the target), I don't think it's necessary to use the `AnnSourceRunningCondition` annotation. Also, I think that handling this in the upload controller with a single annotation is easier and cleaner than adding new logic in the datavolume-controller.

Reply:
Note that a pod not starting due to quota is not a terminal state. In the future someone could add quota and the pod can start. So we should simply indicate in the datavolume condition that source/target pod was not able to start due to quota (or other unknown reasons). If the condition on the DV indicates not enough quota for instance, that gives the user some indication on what to do to fix it. Just saying failed and not doing anything from that point forward defeats the whole eventually consistent idea.

Reply:
You are right, and It's true that using the `AnnSourceRunningCondition` it's the easiest and shortest way to keep the running condition false in the DV without adding new logic. I'll commit the changes soon, but that said, the upload controller will keep overwriting the target PVC running condition, which I'm sure it's not a big deal, but still could lead to some confusion. Lastly, notice that all the error handling to let users know was already implemented in the previous PR: https://github.com/kubevirt/containerized-data-importer/pull/2335 (the event in the DV, changing the running condition, etc.).